webpack.config.js

```diff
@@ -1,207 +1,57 @@
-      'components/Toolbar/Toolbar': 'Toolbar',
-      'containers/FormBuilderLoader/FormBuilderLoader': 'FormBuilderLoader',
-      'deep-freeze-strict': 'DeepFreezeStrict',
-      'graphql-fragments': 'GraphQLFragments',
-      'graphql-tag': 'GraphQLTag',
-      i18n: 'i18n',
-      jquery: 'jQuery',
-      'lib/Backend': 'Backend',
-      'lib/ReducerRegister': 'ReducerRegister',
-      'lib/ReactRouteRegister': 'ReactRouteRegister',
-      'lib/SilverStripeComponent': 'SilverStripeComponent',
-      'page.js': 'Page',
-      'react-addons-test-utils': 'ReactAddonsTestUtils',
-      'react-dom': 'ReactDom',
-      tether: 'Tether',
-      'react-apollo': 'ReactApollo',
-      'react-bootstrap-ss': 'ReactBootstrap',
-      'react-redux': 'ReactRedux',
-      'react-router-redux': 'ReactRouterRedux',
-      'react-router': 'ReactRouter',
-      'react-addons-css-transition-group': 'ReactAddonsCssTransitionGroup',
-      react: 'React',
-      'redux-form': 'ReduxForm',
-      'redux-thunk': 'ReduxThunk',
-      redux: 'Redux',
-      config: 'Config',
-      'lib/Router': 'Router',
-      qs: 'qs',
-      moment: 'moment',
-      modernizr: 'modernizr',
-    },
-    module: {
-      loaders: [
-        {
-          test: /\.js$/,
-          exclude: /(node_modules|thirdparty)/,
-          loader: 'babel',
-          query: {
-            presets: ['es2015', 'react'],
-            plugins: ['transform-object-assign', 'transform-object-rest-spread'],
-            comments: false,
-          },
-        },
-        {
-          test: '/i18n.js/',
-          loader: 'script-loader',
-        },
-        {
-          test: /\.modernizrrc$/,
-          loader: 'modernizr!json',
-        },
-      ],
-    },
-    plugins: [
-      new webpack.ProvidePlugin({
-        jQuery: 'jquery',
-        $: 'jquery',
-      }),
-      new webpack.DefinePlugin({
-        'process.env': {
-          // Builds React in production mode, avoiding console warnings
-          NODE_ENV: JSON.stringify('production'),
-        },
-      }),
-      new webpack.optimize.UglifyJsPlugin({
-        compress: {
-          unused: false,
-          warnings: false,
-        },
-        output: {
-          beautify: false,
-          semicolons: false,
-          comments: false,
-          max_line_len: 200,
-        },
-      }),
-      // Most vendor libs are loaded directly into the 'vendor' bundle (through require()
-      // calls in vendor.js). This ensures that any further require() calls in other
-      // bundles aren't duplicating libs.
-      new webpack.optimize.CommonsChunkPlugin({
-        name: 'vendor',
-        minChunks: Infinity,
-      }),
-    ],
+    devtool: (ENV !== 'production') ? 'source-map' : '',
+    resolve: require('./webpack/js/resolve')(ENV, PATHS),
+    externals: require('./webpack/js/externals')(ENV, PATHS),
+    module: require('./webpack/js/modules')(ENV, PATHS),
+    plugins: require('./webpack/js/plugins')(ENV, PATHS),
```

Comment:
yup, fair call, I'll look at doing that
this was something I had implemented at the start and hadn't revisited since